cmd/cmd.go

```diff
@@ -78,6 +92,22 @@ func ExecInCoreSnap() {
 		}
 	}
 
+	// ensure we do not re-exec into an older version of snapd
+	currentSnapd, err := os.Stat(exe)
+	if err != nil {
+		logger.Noticef("cannot stat %q: %s", exe, err)
+		return
+	}
+	coreSnapSnapd, err := os.Stat(full)
+	if err != nil {
+		logger.Noticef("cannot stat %q: %s", full, err)
+		return
+	}
+	if currentSnapd.ModTime().After(coreSnapSnapd.ModTime()) {
```

Comment:
This is meant as a cheap way to ensure that people get the latest snapd. the problem is of course that if the deb is rebuild it won't reexec into core anymore. I'm not sure that this is actually a problem as usually when people rebuild (or the distro rebuilds) there is a reason (like bugfix in the toolchain) so it seems sensible to use the newer version still. However if you think the behaviour is undesirable I'm happy to remove it again. Should it just unconditionally re-exec? Is that your suggestion?